application/controllers/admin/SurveysGroupsController.php

```diff
@@ -51,69 +55,120 @@ public function create()
+                    );
                 }
-                // What happen iof SurveysGroups saved but no SurveysGroupsettings ?
+                // What happen if SurveysGroups saved but no SurveysGroupsettings ?
             }
         }
-
-        $aData['model'] = $model;
-        $aData['fullpagebar']['savebutton']['form'] = 'surveys-groups-form';
-        $aData['fullpagebar']['returnbutton'] = array(
-            'url'=>'surveyAdministration/listsurveys#surveygroups',
-            'text'=>gT('Close'),
+        $aData = array(
+            'model' => $model,
+            'action' => App()->createUrl("admin/surveysgroups/sa/create", array('#'=>'settingsForThisGroup')),
         );
-
+        $aData['aRigths'] = array(
+            'update' => true,
+            'delete' => false,
+            'owner_id' => true,
+        );
+        $aData['fullpagebar'] = array(
+            'savebutton' => array(
+                'form' => 'surveys-groups-form'
+            ),
+            'returnbutton' => array(
+                'url' => $this->getController()->createUrl('surveyAdministration/listsurveys', array("#"=>'surveygroups')),
+                'text' => gT('Close'),
+            )
+        );
+        /* User for dropdown */
+        $aUserIds = getUserList('onlyuidarray');
+        $userCriteria = new CDbCriteria;
+        $userCriteria->select = array("uid", "users_name", "full_name");
+        $userCriteria->order = "full_name";
+        $userCriteria->addInCondition('uid',$aUserIds);
+        $aData['oUsers'] = User::model()->findAll($userCriteria);
         $this->_renderWrappedTemplate('surveysgroups', 'create', $aData);
     }
 
     /**
-     * Updates a particular model.
+     * Show and updates a particular model.
      * If update is successful, the browser will be redirected to the 'view' page.
      * @param integer $id the ID of the model to be updated
      */
     public function update($id)
     {
         $bRedirect = 0;
         $model = $this->loadModel($id);
-
-        if (isset($_POST['SurveysGroups'])) {
-            $model->attributes = $_POST['SurveysGroups'];
-
-
-                // prevent loop
-                if (!empty($_POST['SurveysGroups']['parent_id'])){
-                    $sgid = $_POST['SurveysGroups']['parent_id'] ;
-                    $ParentSurveyGroup = $this->loadModel($sgid);
-                    $aParentsGsid = $ParentSurveyGroup->getAllParents(true);
-
-                    if ( in_array( $model->gsid, $aParentsGsid  ) ) {
-                        Yii::app()->setFlashMessage(gT("A child group can't be set as parent group"), 'error');
-                        $this->getController()->redirect($this->getController()->createUrl('surveyAdministration/listsurveys').'#surveygroups');
-                    }
+        if (!empty(App()->getRequest()->getPost('SurveysGroups'))) {
+            if (!Permission::model()->hasSurveyGroupPermission($id, 'group', 'update')) {
```

Comment:
Shouldn't this check be at the top of the function?

Reply:
function start at 100, check is done at 103.

I must check if an user can get update with out POST (User can have rights on read part here (unsure)) 

See https://github.com/LimeSurvey/LimeSurvey/blob/6475d0cbc17d3f039398c256c5f0cac7d2569efc/application/controllers/admin/SurveysGroupsController.php#L78

Reply:
Put it in @todo